### DISCUSSION TITLE
mihomo 1.81编译的ppgw貌似没有流量过来。

### DISCUSSION THREAD

**Discussion by Lovest20018** [kkkgo/PaoPaoGateWay#58] [问和答]
刚看到1.81出来了。编译了下，Yacd上面规则都正常显示，节点也能ping通。
但没有任何连接数据。

**Comment by kkkgo** [comment]
经过测试没有问题。建议自己检查配置。
![image](https://github.com/kkkgo/PaoPaoGateWay/assets/4530897/4b914daf-010d-4b40-ab4f-fb055d15c1f6)

**Comment by Lovest20018** [comment]
测试了下。用前置嗅探加物理网卡驱动的ISO是可以正常用的，但如果只是带物理网卡驱动编译的ISO，连接没数据。

**Reply by kkkgo** [reply]
你都能打开yacd就说明网卡正常工作，跟驱动没有关系。如果有嗅探能正常工作你应该考虑你的fakeip的DNS缓存的问题。

**Reply by Lovest20018** [reply]
重启过ppdns了，本地缓存也清除过了-我再测试下。

**Comment by Lovest20018** [comment]
奇怪。我试了mihomo-linux-amd64-compatible-v1.18.1 跟mihomo-linux-amd64-v1.18.1都不行。而且运行的时候都有这个提示
![Snipaste_2024-02-03_20-47-26](https://github.com/kkkgo/PaoPaoGateWay/assets/30313973/e3c04c12-9c5b-4387-988b-3d0f14ff5da6)
换SNIFF=yes编译出来的就是可以。paopaodns升级到最新了。所有设备都重启过-也是一样。

**Reply by kkkgo** [reply]
手动拿一台设备测试，设备的网关地址设置成ppgw的IP地址比如192.168.1.100，设备DNS设置为8.8.8.8，看看能不能用。
不能用的话，再跟踪一下路由，比如跟踪一下google.com的路由，然后看看解析结果，比如`nslookup google.com 192.168.1.100`
再排除一下是否是规则的问题，开启fast_node=yes来测试；然后开启openport=yes，测试socks代理是否能正常用；

**Comment by Lovest20018** [comment]
把网关设置为ppgw的ip的话-nslookup解析出来是fakeip的地址，但还是不能访问，面板里连接也没有任何数据。
开启fast_node=yes的话同样，开启openort=yes的话也是一样的。
我的yaml的规则可能是有些奇怪，按自己使用方式写的。
`proxy-providers:

  🛫 Nexitally:
    type: http
    url: "XXXXXXX"
    path: ./proxies/airport1.yaml


  🛫 Godetia:
    type: http
    url: "XXXXXXX"
    path: ./proxies/airport2.yaml


proxy-groups:

  - {name: 👑 奶昔Pre, type: select, use: [🛫 Nexitally], filter: "^(Hong Kong 22 Premium|.*Premium)"}

  - {name: 🇸🇬 狮城节点, type: select, use: [🛫 Nexitally, 🛫 Godetia], filter: "SG|新加坡|坡|狮城|Singapore"}

  - {name: 🏆 Nexitally, type: select, use: [🛫 Nexitally], exclude-filter: "700.00|Traffic|Expire"}

  - {name: 🏅 Godetia, type: select, use: [🛫 Godetia], exclude-filter: "剩余|到期"}
  
  - {name: 📹 YouTube, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}

  - {name: 📺 Netflix, type: select, proxies: [🇸🇬 狮城节点, 👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}

  - {name: 🐹 Disney+, type: select, proxies: [🇸🇬 狮城节点, 👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}
  
  - {name: 🛩 Telegram, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}

  - {name: 🤖 人工智能, type: select, proxies: [🏆 Nexitally, 👑 奶昔Pre, 🏅 Godetia]}

  - {name: 🎞 Emby, type: select, proxies: [🏅 Godetia, 👑 奶昔Pre, 🏆 Nexitally, DIRECT]}

  - {name: 📢 谷歌服务, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}

  - {name: 🧸 PikPak, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia, DIRECT]}

  - {name: Ⓜ️ 微软服务, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia, DIRECT]}

  - {name: 🎬 国外媒体, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}

  - {name: 🎮 游戏平台, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia, DIRECT]}

  - {name: 🐟 漏网之鱼, type: select, proxies: [👑 奶昔Pre, 🏆 Nexitally, 🏅 Godetia]}

rule-providers:

  google:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Google/Google.yaml"
    path: ./ruleset/google.yaml
    interval: 86400

  game:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Game/Game.yaml"
    path: ./ruleset/game.yaml
    interval: 86400

  youtube:
    type: http
    behavior: domain
    url: "https://cdn.jsdelivr.net/gh/DustinWin/ruleset_geodata@clash/youtube.yaml"
    path: ./ruleset/youtube.yaml
    interval: 86400
 
  netflix:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Netflix/Netflix_Classical.yaml"
    path: ./ruleset/netflix.yaml
    interval: 86400

  Disney:
    type: http
    behavior: domain
    url: "https://cdn.jsdelivr.net/gh/DustinWin/ruleset_geodata@clash/disney.yaml"
    path: ./ruleset/disney.yaml
    interval: 86400

  Telegram:
    type: http
    behavior: ipcidr
    url: "https://cdn.jsdelivr.net/gh/DustinWin/ruleset_geodata@clash/telegramip.yaml"
    path: ./ruleset/telegram.yaml
    interval: 86400

  AI:
    type: http
    behavior: domain
    url: "https://cdn.jsdelivr.net/gh/DustinWin/ruleset_geodata@clash/ai.yaml"
    path: ./ruleset/AI.yaml
    interval: 86400

  microsoft:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/Microsoft/Microsoft.yaml"
    path: ./ruleset/microsoft.yaml
    interval: 86400

  emby:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/Lovest20018/clash-rules/main/Emby.yaml"
    path: ./ruleset/emby.yaml


  pikpak:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/PikPak/PikPak.yaml"
    path: ./ruleset/pikpak.yaml
    interval: 86400

  globalmedia:
    type: http
    behavior: classical
    url: "https://raw.githubusercontent.com/blackmatrix7/ios_rule_script/master/rule/Clash/GlobalMedia/GlobalMedia.yaml"
    path: ./ruleset/globalmedia.yaml
    interval: 86400


rules:
  - RULE-SET,youtube,📹 YouTube
  - RULE-SET,netflix,📺 Netflix
  - RULE-SET,Disney,🐹 Disney+
  - RULE-SET,Telegram,🛩 Telegram
  - RULE-SET,AI,🤖 人工智能
  - RULE-SET,emby,🎞 Emby
  - RULE-SET,pikpak,🧸 PikPak
  - RULE-SET,google,📢 谷歌服务
  - RULE-SET,microsoft,Ⓜ️ 微软服务
  - RULE-SET,globalmedia,🎬 国外媒体
  - RULE-SET,game,🎮 游戏平台
  - MATCH,🐟 漏网之鱼`
  
ppgw.ini的配置
`#paopao-gateway

# mode=socks5|ovpn|yaml|suburl|free
# default: free
mode=yaml

# Set fakeip's CIDR here
# default: fake_cidr=7.0.0.0/8
fake_cidr=7.0.0.0/8

# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=192.168.1.251
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304

# Clash's web dashboard
clash_web_port="80"
clash_web_password="811215"

# defaultя╝Ъopenport=no
# socks+http mixed 1080
openport=no

# default: udp_enable=no
udp_enable=yes

# default:30
sleeptime=30

# socks5 mode settting
# default: socks5_ip=gatewayIP
socks5_ip="10.10.10.5"
# default: socks5_port="7890"
socks5_port="7890"

# ovpn mode settting
# The ovpn file in the same directory as the ppgw.ini.
# default: ovpnfile=custom.ovpn
ovpnfile="custom.ovpn"
ovpn_username=""
ovpn_password=""

# yaml mode settting
# The yaml file in the same directory as the ppgw.ini.
# default: yamlfile=custom.yaml
yamlfile="custom.yaml"

# suburl mode settting
suburl="https://..."
subtime=1d

# fast_node=check/yes/no
fast_node=no
test_node_url="https://www.youtube.com/generate_204"
ext_node="Traffic|Expire| GB|Days|Date"
cpudelay="3000"

# dns burn setting
# depand on fast_node=yes & mode=suburl/yaml
dns_burn=no
# If used with PaoPaoDNS, you can set the PaoPaoDNS:53
ex_dns="192.168.1.251:53"

# Network traffic records
net_rec=yes
max_rec=5000`

之前用任何mihomo的内核编译都没问题的，就是在1.81这样-只能编译嗅探iso才可以。

**Reply by kkkgo** [reply]
那你编译嗅探的先用着吧，我有空再测试下。其实嗅探的体验用起来会更好。

**Comment by kkkgo** [comment]
应该是这个问题，我看到有其他人反馈了issue。
https://github.com/MetaCubeX/mihomo/issues/1027

**Comment by Lovest20018** [comment]
感谢大佬-mihomo那边修复了。
